dallinger/experiment_server/sockets.py

```diff
@@ -87,8 +89,23 @@ def stop(self):
 
 @sockets.route('/receive_chat')
 def outbox(ws):
-    chat_backend.subscribe(ws)
+    """Sends outgoing messages to subscribed clients"""
+    channel = request.args.get('channel')
+    chat_backend.subscribe(ws, channel=channel)
 
     while not ws.closed:
         # Wait for chat backend
         gevent.sleep(0.1)
+
+
+@sockets.route('/send_chat')
+def inbox(ws):
+    """Receives incoming messages and inserts them into a Redis channel"""
+    channel = request.args.get('channel')
+
+    while not ws.closed:
+        # Sleep to prevent *constant* context-switches.
+        gevent.sleep(0.1)
```

Comment:
This I copied from Heroku article you'd cited (https://devcenter.heroku.com/articles/python-websockets). I'd rate my understanding of how greenlets work at about 65%, so I'll defer to you (not sure how I'd determine if it was helping vs. hurting).